### ISSUE TITLE
Error when horizontal scroll appears with Mantine 6.0.18

### ISSUE THREAD

**Issue by rolderman** [icflorescu/mantine-datatable#398]
**Describe the bug**
`TypeError: Cannot read properties of undefined (reading 'scrollLeft')`
This error happens when horizontal scrollbar appears. In both cases: when changed veiwport width and when initial width of table greater parent width.

**To Reproduce**
1. Upgrade to Mantine 6.0.18
2. Create table with horizontal scrollbar

**Desktop (please complete the following information):**

- OS: MacOS 12.6.7
- Browser Chrome
- Version 115.0.5790.114

**Comment by zeyadetman** [comment]
I have the same error even with lower versions of Mantine.

**Comment by rolderman** [comment]
6.0.17 works fine

**Comment by KauaTec4You** [comment]
What version of mantine-datatable did you use to work with @mantine/core 6.0.17?

**Comment by rolderman** [comment]
Latest 2.9.7

**Comment by Felistus** [comment]
I have tried tried all here but I still get same error. here is a sample of my package.json file.
```
"@mantine/core": "^6.0.16",
    "@mantine/dates": "^6.0.16",
    "@mantine/form": "^6.0.16",
    "@mantine/hooks": "^6.0.16",
    "@mantine/modals": "^6.0.16",
    "@mantine/next": "^6.0.16",
    "@mantine/notifications": "^6.0.16",
    "mantine-datatable": "^2.9.3"
```

**Comment by icflorescu** [comment]
This is due to https://github.com/mantinedev/mantine/pull/4512

**Comment by icflorescu** [comment]
@Felistus @zeyadetman
To lock a dependency to a **specific** version in `package.json`, you have to use this syntax: `"@mantine/core": "6.0.17"`.
AFAIK, `"@mantine/core": "^6.0.16"` **means `6.0.16` or a newer** patch version, which usually results in downloading & using the latest version. Which is not what you want in this case.

**Comment by icflorescu** [comment]
@Felistus

> how does this relate to the issue please? TypeError: Cannot read properties of undefined (reading 'scrollLeft')
> I am new to this tool and the explanation makes no clear sense to me. Please assist

Mantine DataTable is using the [Mantine `useElementSize` hook](https://mantine.dev/hooks/use-element-size/) internally to generate element references.

In Mantine `6.0.18`, [a PR unpredictably changed the signature of `useElementSize`](https://github.com/mantinedev/mantine/pull/4512). The `ref` returned by the function changed from `MutableRefObject<T>` to `(node: T | null) => void`, which, **if absolutely necessary**, should have been done in a major version release, not a patch.

Before accepting the PR, the `MutableRefObject` `ref` had a `.current` property, which, in its turn, had a `.scrollLeft` property.
After the PR, the `ref` has no `.current` property, which, as expected, results in this error: `Cannot read properties of undefined (reading 'scrollLeft')`.

Hope this makes it clearer.

The next patch of Mantine [should fix the issue](https://github.com/mantinedev/mantine/pull/4512#issuecomment-1671879404).
**Until then, make sure to lock Mantine's version to 6.0.17 in your project**.

**Comment by Felistus** [comment]
> @Felistus
> 
> > how does this relate to the issue please? TypeError: Cannot read properties of undefined (reading 'scrollLeft')
> > I am new to this tool and the explanation makes no clear sense to me. Please assist
> 
> Mantine DataTable is using the [Mantine `useElementSize` hook](https://mantine.dev/hooks/use-element-size/) internally to generate element references.
> 
> In Mantine `6.0.18`, [a PR unpredictably changed the signature of `useElementSize`](https://github.com/mantinedev/mantine/pull/4512). The `ref` returned by the function changed from `MutableRefObject<T>` to `(node: T | null) => void`, which, **if absolutely necessary**, should have been done in a major version release, not a patch.
> 
> Before accepting the PR, the `MutableRefObject` `ref` had a `.current` property, which, in its turn, had a `.scrollLeft` property. After the PR, the `ref` has no `.current` property, which, as expected, results in this error: `Cannot read properties of undefined (reading 'scrollLeft')`.
> 
> Hope this makes it clearer.
> 
> The next patch of Mantine [should fix the issue](https://github.com/mantinedev/mantine/pull/4512#issuecomment-1671879404). **Until then, make sure to lock Mantine's version to 6.0.17 in your project**.

Thank you so much Ionut-Cristian for this wonderful explanation. I really appreciate this. 
Now I have a good understanding of what's happening.

**Comment by icflorescu** [comment]
This was [promptly fixed in Mantine `6.0.19`](https://github.com/mantinedev/mantine/pull/4512#issuecomment-1672932954) - many thanks to @rtivital 🙏, so please make sure to update both `mantine-datatable` and `@mantine/*` to the latest version.
